internal/manifests/gateway/build.go

```diff
@@ -106,11 +106,22 @@ const (
 	portPublic   = 8080
 )
 
+func getTenantHeaderFlag(t *tempov1alpha1.TenantsSpec) []string {
```

Comment:
If this is a single flag... I would expect to get only one `string` returned, not a slice. WDYT?

Reply:
You're returning a `nil` or `1-element` slice. You can solve that with
```go
headerFlags := []string{getTenantHeaderFlag(params.Tempo.Spec.Tenants)}
```

Reply:
How would the function signature of `getTenantHeaderFlag` look like?

Reply:
We should decide how to handle these `if-else` statements. 

The original idea was to use `mergo` and just merge manifests. I would rather stick to a single more readable approach.

Reply:
works for me too. So it would look like this:
```golang
patch := configureGatewayTenant(...)

if err := mergo.Merge(origin, patch, mergo.WithAppendSlice); err != nil {
	return nil, fmt.Errorf("failed to merge multi tenant settings into gateway container spec %w", err)
}
```